src/System Application/App/AI/src/Azure OpenAI/Operation Response/AOAIFunctionResponse.Codeunit.al

```diff
@@ -0,0 +1,91 @@
+// ------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+// ------------------------------------------------------------------------------------------------
+namespace System.AI;
+
+/// <summary>
+/// The status and result of an functions.
+/// </summary>
+codeunit 7758 "AOAI Function Response"
+{
+    Access = Public;
+    InherentEntitlements = X;
+    InherentPermissions = X;
+
+    var
+        Success: Boolean;
+        FunctionCall: Boolean;
+        FunctionName: Text;
+        Result: Variant;
+        Error: Text;
+        ErrorCallStack: Text;
+
+    /// <summary>
+    /// Get whether the function call was successful.
+    /// </summary>
+    /// <returns>True if the call was successful, false otherwise.</returns>
+    procedure IsSuccess(): Boolean
+    begin
+        exit(Success);
+    end;
+
+    /// <summary>
+    /// Get the return value of the function that was called.
+    /// </summary>
+    /// <returns>The return value from the function</returns>
+    procedure GetResult(): Variant
+    begin
+        exit(Result);
+    end;
+
+    /// <summary>
+    /// Get the error message from the function that was called.
+    /// </summary>
+    /// <returns>The error message from the function.</returns>
+    procedure GetError(): Text
+    begin
+        exit(Error);
+    end;
+
+    /// <summary>
+    /// Get whether the operation was a function call.
+    /// </summary>
+    /// <returns>True if it was a function call, false otherwise.</returns>
+    procedure IsFunctionCall(): Boolean
```

Comment:
Make this internal. No reason for accessing directly from here. It should be accessed from OperationResponse only.